os/arch/arm/include/armv7-m/irq.h

```diff
@@ -91,6 +91,23 @@
 #define CONFIG_SYS_NNEST 2
 #endif
 
+/* MPU Region Definitions */
+
+#define MPU_REG0       (0)
+#define MPU_REG1       (1)
+#define MPU_REG2       (2)
+#define MPU_REG3       (3)
+#define MPU_REG4       (4)
+#define MPU_REG5       (5)
+#define MPU_REG6       (6)
+#define MPU_REG7       (7)
+#if (CONFIG_ARMV7M_MPU_NREGIONS == 12)
+#define MPU_REG8       (8)
+#define MPU_REG9       (9)
+#define MPU_REG10      (10)
+#define MPU_REG11      (11)
```

Comment:
What if the number of mpu regions is greater than 12 (or equal to 16 as mentioned in defconfig)?

Reply:
modified and added changes for more mpu regions in case of M7